core/node/storage/pg_stream_store.go

```diff
@@ -135,22 +154,38 @@ func (s *PostgresStreamStore) txRunnerWithUUIDCheck(
 	)
 }
 
-// processErrorForMissingPartition looks for the specific error type returned when a table is not found
-// and returns that as a NOT_FOUND river error for the specified stream.
-func processErrorForMissingPartition(err error, streamId StreamId) error {
-	var pgErr *pgconn.PgError
-	if errors.As(err, &pgErr) {
-		if pgErr.Code == pgerrcode.UndefinedTable {
-			return RiverError(Err_NOT_FOUND, "stream not found in local storage", "streamId", streamId)
-		}
-	}
-	return err
+// createPartitionSuffix determines the partition mapping for a particular stream id the
+// hex encoding of the first byte of the xxHash of the stream ID.
+func createPartitionSuffix(streamId StreamId, reducedParitions bool) string {
+	// Media streams have separate partitions to handle the different data shapes and access
+	// patterns. The partition suffix is prefixed with an "m". Regular streams are assigned to
+	// partitions prefixed with "r", e.g. "miniblocks_ra4".
+	streamType := "r"
+	if streamId.Type() == STREAM_MEDIA_BIN {
+		streamType = "m"
+	}
+
+	// Do not hash the stream bytes directly, but hash the hex encoding of the stream id, which is
+	// what we store in the database. This leaves the door open for installing xxhash on postgres
+	// and debugging this way in the future.
+	hash := xxhash.Sum64String(streamId.String())
+	// For test installations, expect 4 partitions
+	if reducedParitions {
+		bt := hash % 4
+		return fmt.Sprintf("%v%02x", streamType, bt)
+	}
+	return fmt.Sprintf("%v%016x", streamType, hash)[:3]
 }
 
 // sqlForStream escapes references to partitioned tables to the specific partition where the stream
 // is assigned whenever they are surrounded by double curly brackets.
-func sqlForStream(sql string, streamId StreamId) string {
-	suffix := createTableSuffix(streamId)
+func (s *PostgresStreamStore) sqlForStream(sql string, streamId StreamId, migrated bool) string {
+	var suffix string
+	if migrated {
+		suffix = createPartitionSuffix(streamId, s.config.TestMode)
+	} else {
+		suffix = createTableSuffix(streamId)
+	}
 
 	sql = strings.ReplaceAll(
```

Comment:
Not related to this PR but it would be nice to replace the 3 `ReplaceAll` calls with `strings.NewReplacer` that allows to do multiple replacements at once.

Reply:
This was actually tested with go benchmark and for this use case strings.NewReplacer is several times slower and allocates way more.